cmd/zc_filter.go

```diff
@@ -357,6 +357,54 @@ func (_ IncludeBeforeDateFilter) FormatAsUTC(t time.Time) string {
 	return formatAsUTC(t)
 }
 
+type softDeleteFilter struct {
+	deleteSnapshots bool
+	deleteVersions  bool
+}
+
+func (s *softDeleteFilter) DoesSupportThisOS() (msg string, supported bool) {
+	msg = ""
+	supported = true
+	return
+}
+
+func (s *softDeleteFilter) AppliesOnlyToFiles() bool {
+	return false
+}
+
+func (s *softDeleteFilter) DoesPass(storedObject StoredObject) bool {
+	if s.deleteVersions && s.deleteSnapshots {
```

Comment:
This seems like it could all be simplified down into a single conditional.

e.g. `return (s.deleteVersions || s.deleteSnapshots) && ((s.deleteVersions && storedObject.blobVersionID != "") || (s.deleteSnapshots && storedObject.blobSnapshotID != ""))`